components/ui/toast/use-toast.tsx

```
@@ -0,0 +1,222 @@
+const listeners: Array<(state: State) => void> = []
+
+let memoryState: State = { toasts: [] }
+
+function dispatch(action: Action) {
+  memoryState = reducer(memoryState, action)
+  listeners.forEach((listener) => {
+    listener(memoryState)
+  })
+}
+
+interface Toast extends Omit<ToasterToast, "id"> {}
```

Biome (2.1.2)

[error] Shouldn't redeclare 'Toast'. Consider to delete it or rename it.

'Toast' is defined here:


(lint/suspicious/noRedeclare)